src/main/java/hudson/plugins/sonar/action/SonarAnalysisAction.java

```diff
@@ -117,4 +146,11 @@ public String getInstallationName() {
   public String getCredentialsId() {
     return credentialsId;
   }
+
+  @CheckForNull
+  @Transient
+  public String getUsableServerUrl() {
```

Comment:
I might name it `getServerDisplayUrl()` or maybe just `getDisplayUrl()`?

Reply:
> I might name it getServerDisplayUrl() or maybe just getDisplayUrl()?

It's the opposite: what I've called the "usable" server URL is the URL Jenkins should actually "use" (the right URL for calling SQ webservices from Jenkins), which might be different from the public SQ URL (the one you would display to the end-user). It's the one you would configure in your `SonarInstallation` ("SonarQube servers" in the Jenkins global configuration).

But yes, I'm not super happy with this name either. I didn't want to call it "internal URL", because on m setup it's just the same as the public URL, so it's not always "internal". Should really be `getTheRightBaseURLForCallingSQWebservices()` :)
I'm open to better suggestions, of course.

Reply:
What about `getServerApiEndpoint()`?

Reply:
Ah, it's clear now. In that case, I'd call it `getServiceUrl()`, though maybe too close to "ServerUrl"?

Reply:
Thanks @henryju, it's now `getServerApiEndpoint()`: 3b28a81